From 8a1d66c7aec020f3e90254ed2fa55ecd9494fcc3 Mon Sep 17 00:00:00 2001
From: Michael Brown <mcb30@ipxe.org>
Date: Sat, 27 Jun 2020 20:43:32 +0100
Subject: [PATCH] [golan] Add explicit type casts for nodnic_queue_pair_type

GCC 10 emits warnings for implicit conversions of enumerated types.

The flexboot_nodnic code defines nodnic_queue_pair_type with values
identical to those of ib_queue_pair_type, and implicitly casts between
them.  Add an explicit cast to fix the warning.

Signed-off-by: Michael Brown <mcb30@ipxe.org>
---
 src/drivers/infiniband/flexboot_nodnic.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/drivers/infiniband/flexboot_nodnic.c b/src/drivers/infiniband/flexboot_nodnic.c
index 93bb05446..7d039fffc 100644
--- a/src/drivers/infiniband/flexboot_nodnic.c
+++ b/src/drivers/infiniband/flexboot_nodnic.c
@@ -365,7 +365,8 @@ static int flexboot_nodnic_create_qp ( struct ib_device *ibdev,
 		goto qp_alloc_err;
 	}
 
-	status = nodnic_port_create_qp(&port->port_priv, qp->type,
+	status = nodnic_port_create_qp(&port->port_priv,
+			(nodnic_queue_pair_type) qp->type,
 			qp->send.num_wqes * sizeof(struct nodnic_send_wqbb),
 			qp->send.num_wqes,
 			qp->recv.num_wqes * sizeof(struct nodnic_recv_wqe),
@@ -406,7 +407,8 @@ static void flexboot_nodnic_destroy_qp ( struct ib_device *ibdev,
 	struct flexboot_nodnic_port *port = &flexboot_nodnic->port[ibdev->port - 1];
 	struct flexboot_nodnic_queue_pair *flexboot_nodnic_qp = ib_qp_get_drvdata ( qp );
 
-	nodnic_port_destroy_qp(&port->port_priv, qp->type,
+	nodnic_port_destroy_qp(&port->port_priv,
+			(nodnic_queue_pair_type) qp->type,
 			flexboot_nodnic_qp->nodnic_queue_pair);
 
 	free(flexboot_nodnic_qp);
